internal/utils_fs.go

```diff
@@ -13,24 +13,41 @@ import (
 
 // getRandomDirectory creates a random directory in /tmp, creates the directories and returns the full path
 // directory is of the form `/tmp/<random-word>/<random-word>/<random-word>`
-func getRandomDirectory(stageHarness *test_case_harness.TestCaseHarness) (string, error) {
+// If performCleanup is true, the directory will be cleaned up when the test is completed
+// The total possible directories is 10^3 = 1000
+// This can be used without cleanup in most cases
+func getRandomDirectory(stageHarness *test_case_harness.TestCaseHarness, performCleanup bool) (string, error) {
```

Comment:
I don't think there should be a case ever where we don't want cleanup 🤔 Comment added above